fieldsolver/gridGlue.cpp

```diff
@@ -179,6 +190,210 @@ void feedMomentsIntoFsGrid(dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>&
 
   MPI_Waitall(sendRequests.size(), sendRequests.data(), MPI_STATUSES_IGNORE);
 
+  // ---------------------------------------------------------------------------------
+  // Boxcar Smoothening
+  const int *mntDims= &momentsGrid.getLocalSize()[0];
+  std::array<Real,fsgrids::moments::N_MOMENTS> *Grid;
+  std::array<Real,fsgrids::moments::N_MOMENTS> *GridL1;
+  std::array<Real,fsgrids::moments::N_MOMENTS> *GridL2;
+  std::array<Real,fsgrids::moments::N_MOMENTS> *GridR1;
+  std::array<Real,fsgrids::moments::N_MOMENTS> *GridR2;
+  std::array<Real,fsgrids::moments::N_MOMENTS> *swap;
+
+
+  /*Kernel Weights are fractional numbers 
+
+              std::array weights[i]
+    weight= ---------------------------
+                2*refLevel+1
+
+  and are calculated as shown above while iterating over each dimensions
+  The numerator is calculated filling the weights array
+  and the denominator depends only on the stencil width. 
+  */
+  float weightC, weightL1, weightL2, weightR1, weightR2;
+  int blurPasses = 3;
+  int refLevel;
+  std::array<int, 5> weights;
```

Comment:
Is 5 a constant? Or does it depend on the maximum refinement? See comments below too.